[페이지] CommunityComment

Ho2eny opened the pull request:
### 요구 사항
| Community  | Community 2 |
|-|-|
|![image](https://user-images.githubusercontent.com/50590025/193588917-5e3334ee-585a-4b73-b5ab-10a451391de8.png)|![image](https://user-images.githubusercontent.com/50590025/193588993-13a251bb-9e44-4379-9ae4-70ee24a3e8c8.png)|

### 수정 내역

### 당부의 말씀

### 테스트 결과
![image](https://user-images.githubusercontent.com/50590025/193589110-6161d071-a3d3-468e-a292-68c6b0990910.png)

Ho2eny committed:
feat: CommunityCommentFooter 컴포넌트 추가

Ho2eny committed:
feat: CommunityCommentItem 컴포넌트 추가

Ho2eny committed:
feat: CommunityComment 페이지 추가

Ho2eny committed:
feat: CommunityCommentReadMore 컴포넌트 추가

답글 더보기 컴포넌트 추가

wlsdhr0831 commented on src/pages/CommunityComment/components/CommunityCommentReadMore/CommunityCommentReadMore.view.tsx:
```diff
@@ -0,0 +1,29 @@
+          -답글 접기
+        </span>
+      )}
+      {!isOpen && <span onClick={onClick}>-답글 {replyCount}개 더 보기</span>}
```
```
{ isOpen ? <></> : <></>}
```

이렇게 하는 것이 더 좋을 것 같습니다~

Ho2eny replied:
https://github.com/TheNewestWorld/groove-front/pull/60/commits/8d979de3226351dd93114d356bd901cee712d082 수정했습니다

wlsdhr0831 commented on src/pages/CommunityComment/components/CommunityCommentReadMore/CommunityCommentReadMore.view.tsx:
```diff
@@ -0,0 +1,29 @@
+  isOpen,
+  replyCount,
+  className,
+  onClick,
```
해당 컴포넌트에서는 onClick 하나로 하는게 아니라
지금 댓글열기 액션인지, 댓글접기 액션인지 따로따로 정의해서 사용하는게
명확할 것 같습니다~

Ho2eny replied:
아 그렇네요, onClick을 전달해줘야 하는 쪽에서는 명확히 현재 상태를 알기가 불편하겠네요 👍

Ho2eny replied:
https://github.com/TheNewestWorld/groove-front/pull/60/commits/f541bf573b47b7c68db63bba66bec980e5e3965f 반영했습니다~

wlsdhr0831 commented on src/pages/CommunityComment/components/CommunityCommentReadMore/CommunityCommentReadMore.module.scss:
```diff
@@ -0,0 +1,12 @@
+
+  .open {
+    margin-left: 42px;
+  }
```
모듈 이라 해당 괄호 안에 정의하지 않아도 될 것 같아요~

Ho2eny replied:
https://github.com/TheNewestWorld/groove-front/pull/60/commits/dff5253d42338692c69848976aa02db99637e66d 수정했습니다

wlsdhr0831 commented on src/pages/CommunityComment/components/CommunityCommentItem/CommunityCommentItem.view.tsx:
```diff
@@ -0,0 +1,70 @@
+interface TagInfo {
+  nickname: string;
+  profileUrl: string;
+}
```
해당 타입을 중복으로 여러곳에서 사용하는 것이 아니라면
인라인으로 적어주는 것이 파일내를 돌아다니며 타입을 파악하지 않아도 되서
더 선호하는 편입니다~

wlsdhr0831 replied:
그리고 태그된 사람의 프로필 url 이 해당 유저의 개인 페이지로 이동하는것이 목적이라면
그냥 그 유저의 고유 id 값을 받아서 그걸 가지고 페이지 전환을 하면 되지 않을까 싶습니다~

개인 페이지라면 고정된 기본 url이 있고 거기서 id 값을 가지고 변형을 줄테니까요??
ex) github.com/{id}

Ho2eny replied:
https://github.com/TheNewestWorld/groove-front/pull/60/commits/afb93bfa05d01898258b44a9b78e40e39bb2106c 반영하였습니다.
inline으로 표현하도록 변경하였고, `profileUrl`이 아닌 `userId`를 받도록 변경하였습니다.
마이페이지의 주소가 아직 정해지지 않아, 추후에 적용할 수 있도록 커멘트 남겼습니다

wlsdhr0831 commented on src/pages/CommunityComment/components/CommunityCommentItem/CommunityCommentItem.view.tsx:
```diff
@@ -0,0 +1,70 @@
+  className,
+  onCommenterClick,
+  onReplyClick,
+  onOptionClick,
```
`commenter` 라는 단어보다 `writer` 라는 단어는 어떠신가요
결국 댓글로 하나의 글이고 그걸 작성한 사람이다.. 요렇게?
아니면 아이디가 아니라 닉네임이다 라는걸 명확하게 표현해도 좋을 것 같아요

wlsdhr0831 replied:
`commenterImage` 도 `profileImage` 라고 사용해도 해당 댓글 작성자의 프로필 사진이구나 싶을 것 같은데
어떠신가요~? 해당 이미지가 어떠한 정보를 가지고 있는건지 좀 더 쉽게 파악할 수 있을 것 같습니다!

wlsdhr0831 replied:
`elpasedTime` 은 위에서 계산해서 주는게 아니라 여기 코멘트 컴포넌트는 작성한 시간 자체를 받아
계산해서 보여주는게 좀 더 맞을 것 같아요!

나중에 보여지는 방식이 변하게 되면 요 컴포넌트를 왔다가 얘를 사용하는 컴포넌트를 거슬러 올라가서
데이터 포맷을 변경해야하니까..??

wlsdhr0831 replied:
`optionClick`으로 인해 작은 창이 나타나고 사라지는건 CommentItem 컴포넌트에서 관리해야하는것 같고,
그 옵션들 안에서 진행할 액션들의 이벤트가 있는게 맞는 것 같습니다~

옵션들이 정해지지 않은 거라면 TODO로 설명해두고 넘기시는게 어떤가요~?

Ho2eny replied:
> commenter 라는 단어보다 writer 라는 단어는 어떠신가요

https://github.com/TheNewestWorld/groove-front/pull/60/commits/bcd547022ddc2da5c0ee9a0e9ab333f84b7eac41 반영했습니다

> commenterImage 도 profileImage 라고 사용해도 해당 댓글 작성자의 프로필 사진이구나 싶을 것 같은데
어떠신가요~?

https://github.com/TheNewestWorld/groove-front/pull/60/commits/bcd547022ddc2da5c0ee9a0e9ab333f84b7eac41 반영했습니다.

의견 감사합니다! 👍

Ho2eny replied:
> elpasedTime 은 위에서 계산해서 주는게 아니라 여기 코멘트 컴포넌트는 작성한 시간 자체를 받아
계산해서 보여주는게 좀 더 맞을 것 같아요!

>나중에 보여지는 방식이 변하게 되면 요 컴포넌트를 왔다가 얘를 사용하는 컴포넌트를 거슬러 올라가서
데이터 포맷을 변경해야하니까..??

https://github.com/TheNewestWorld/groove-front/pull/60/commits/086bafac6de9ab8a6f3ad65c09abe5f07f8726e6 반영했습니다

Ho2eny replied:
> optionClick으로 인해 작은 창이 나타나고 사라지는건 CommentItem 컴포넌트에서 관리해야하는것 같고,
그 옵션들 안에서 진행할 액션들의 이벤트가 있는게 맞는 것 같습니다~

https://github.com/TheNewestWorld/groove-front/pull/60/commits/136f651207b7e344e00a24eb6e4d5371ab6a73c7 반영했습니다, 확인 부탁드려요!!

wlsdhr0831 commented on src/pages/CommunityComment/components/CommunityCommentItem/CommunityCommentItem.view.tsx:
```diff
@@ -0,0 +1,70 @@
+        src={commenterImage}
+        onClick={onCommenterClick}
+      />
+      <div className={styles.containerBody}>
```
content 라는 클래스명은 어떠세요~?

Ho2eny replied:
https://github.com/TheNewestWorld/groove-front/commit/bcd547022ddc2da5c0ee9a0e9ab333f84b7eac41 반영했습니다!

wlsdhr0831 commented on src/pages/CommunityComment/components/CommunityCommentItem/CommunityCommentItem.view.tsx:
```diff
@@ -0,0 +1,70 @@
+            {taggedUsers &&
+              taggedUsers.map(item => (
+                <a href={item.profileUrl} className={styles.tag}>
+                  @{item.nickname}{" "}
```
뒤에 공백은 실수일까요~?
태그된 닉네임들간의 간격을 의도한거라면 tag 클래스로 css 설정하면 되지 않을까 싶습니다!

Ho2eny replied:
https://github.com/TheNewestWorld/groove-front/pull/60/commits/dff5253d42338692c69848976aa02db99637e66d 반영했습니다

wlsdhr0831 commented on src/pages/CommunityComment/components/CommunityCommentItem/CommunityCommentItem.view.tsx:
```diff
@@ -0,0 +1,70 @@
+              ))}
+            <span className={styles.comment}>{comment}</span>
+          </div>
+          <div className={styles.dots}>
```
option 이라는 클래스명은 어떠세요~?

Ho2eny replied:
https://github.com/TheNewestWorld/groove-front/commit/bcd547022ddc2da5c0ee9a0e9ab333f84b7eac41 반영했습니다!

wlsdhr0831 commented on src/pages/CommunityComment/components/CommunityCommentFooter/CommunityCommentFooter.view.tsx:
```diff
@@ -0,0 +1,32 @@
+  onSubmitComment: (comment: string) => void;
+}
+
+const CommunityCommentFooter = ({ onSubmitComment }: Props) => {
```
Footer 라는 단어보다 Input 이라고 하는건 어떤가요~?
위치보다 해당 컴포넌트의 역할이 좀 더 강조되게요!

wlsdhr0831 replied:
그리고 textarea 보다 input 태그가 좀 더 나은 것 같습니다!
textarea는 기본적으로 사이즈 조절도 되기 때문에 추가 설정해야하는게 너무 많아져요...

Ho2eny replied:
> 그리고 textarea 보다 input 태그가 좀 더 나은 것 같습니다!
textarea는 기본적으로 사이즈 조절도 되기 때문에 추가 설정해야하는게 너무 많아져요..

2줄 처리하는게 조금 까다로워서 별도의 PR에서 다시 작업을 해보겠습니다

Ho2eny replied:
>> 그리고 textarea 보다 input 태그가 좀 더 나은 것 같습니다!
textarea는 기본적으로 사이즈 조절도 되기 때문에 추가 설정해야하는게 너무 많아져요..

> 2줄 처리하는게 조금 까다로워서 별도의 PR에서 다시 작업을 해보겠습니다

https://github.com/TheNewestWorld/groove-front/pull/60/commits/dff5253d42338692c69848976aa02db99637e66d 반영했습니다

Ho2eny replied:
> Footer 라는 단어보다 Input 이라고 하는건 어떤가요~?

https://github.com/TheNewestWorld/groove-front/pull/60/commits/39b6bfe857bcfa0b10efa15033c7b260d54b6fc8 반영했습니다!

wlsdhr0831 commented on src/pages/CommunityComment/components/CommunityCommentFooter/CommunityCommentFooter.view.tsx:
```diff
@@ -0,0 +1,32 @@
+
+const CommunityCommentFooter = ({ onSubmitComment }: Props) => {
+  const [comment, setComment] = useState<string>("");
+  return (
```
return 윗줄은 빈 줄이 있는게 국룰의 컨벤션 입니당 ㅎㅎ

wlsdhr0831 replied:
return 을 기준으로 위는 로직, 아래는 뷰 이렇게 구분이 되니까요..??

Ho2eny replied:
**return의 미학이군요**

Ho2eny replied:
https://github.com/TheNewestWorld/groove-front/pull/60/commits/8d979de3226351dd93114d356bd901cee712d082 수정했습니다

wlsdhr0831 commented on src/pages/CommunityComment/CommunityComment.view.tsx:
```diff
@@ -0,0 +1,90 @@
+export interface CommentPair {
+  comment: CommunityCommentItemType;
+  replies: CommunityCommentItemType[];
+}
```
재사용 되는 타입이 아니라면 인라인하기~

Ho2eny replied:
https://github.com/TheNewestWorld/groove-front/pull/60/commits/afb93bfa05d01898258b44a9b78e40e39bb2106c 반영했습니다! 👍

wlsdhr0831 commented on src/pages/CommunityComment/CommunityComment.view.tsx:
```diff
@@ -0,0 +1,90 @@
+
+const CommunityComment = ({
+  comments = [],
+  onCloseClick,
```
onClick~, onFocus~, onOpen~ 이렇게 네이밍을 하는게 더 좋은 것 같아요!

wlsdhr0831 commented on src/pages/CommunityComment/CommunityComment.view.tsx:
```diff
@@ -0,0 +1,90 @@
+          onRightClick={onCloseClick}
+        />
+        {comments.map((item, index) => (
+          <div>
```
반복문에는 key 값이 필요합니다~

wlsdhr0831 replied:
고유한 값이여야 해서 comment의 id 값이 적절할 듯 해요~

wlsdhr0831 commented on src/pages/CommunityComment/CommunityComment.view.tsx:
```diff
@@ -0,0 +1,90 @@
+            <CommunityCommentItem {...item.comment} />
+
+            {item.replies &&
+              item.replies.length > 0 &&
```
`item.replies?.length > 0` 이렇게 한번에 가능할 것 같아요!

Ho2eny replied:
https://github.com/TheNewestWorld/groove-front/pull/60/commits/8d979de3226351dd93114d356bd901cee712d082 수정했습니다

wlsdhr0831 commented on src/pages/CommunityComment/CommunityComment.view.tsx:
```diff
@@ -0,0 +1,90 @@
+      isOpen: true,
+      limit: 10,
+    }),
+  );
```
이걸 여기서 관리하는게 맞나 싶네용..

wlsdhr0831 replied:
흐음... 저라면..!
`CommunityComment -> CommunityCommentList` : 컴포넌트 명 수정
`51 - 82줄 -> CommunityComment` : 컴포넌트 분리
라고 컴포넌트 추가해서 그 안에서 하나의 댓글에 대한 댓글 열기, 접기 데이터를 관리하려고 할 것 같아요!

Ho2eny replied:
https://github.com/TheNewestWorld/groove-front/pull/60/commits/1a768482ee8054acb9850c18b9c0f0ec810091cd 반영하였습니다

wlsdhr0831 reviewed (commented):
양이 많아서 그런지 리뷰가 좀 많네요..ㅎㅎ

무조건 반영하실 필요는 없고, 코드를 짠 사람이 아닌 제 3자가 봤을 때
좀 더 이해에 도움이 될 것 같은 방향으로 제안해보았습니답

wlsdhr0831 commented on src/pages/CommunityComment/CommunityComment.view.tsx:
```diff
@@ -0,0 +1,90 @@
+}
+
+export interface Props {
+  comments?: CommentPair[];
```
문득 드는 생각이 여기 컴포넌트에서는 List가 옵셔날 한게 아니라
필수 값으로 있도록 해야할 것 같아요~

댓글이 없으면 작성된 댓글이 없다는 화면이 보여지고,
댓글이 있으면 이 컴포넌트가 보여지도록!
그래서 이 컴포넌트는 List 객체가 항상 있는게 낫지 않나 싶네용~

Ho2eny committed:
bugfix: 표현법을 수정합니다

true|false 처리를 `:` 로 변경
nullable 체크를 `?`로 변경

Ho2eny committed:
bugfix: 빌드 에러 수정

Ho2eny committed:
bugfix: 스타일을 변경합니다

- module.scss에서 중첩된 선언 제거
- 불필요한 공백 추가를 스타일로 변경
- textarea 변경 필요 커멘트

Ho2eny committed:
bugfix: 재사용되지 않는 타입 inline으로 변경

- 재사용되지 않는 타입을 inline으로 표현합니다.

Ho2eny committed:
fix: 변수 이름을 변경합니다

의미있는 이름이 되도록 변경합니다

Ho2eny committed:
fix: CommunityCommentFooter의 이름을 변경합니다

CommunityCommentFooter를 CommunityCommentInput으로 변경합니다

wlsdhr0831 commented on src/pages/CommunityComment/components/CommunityCommentItem/CommunityCommentItem.stories.tsx:
```diff
@@ -27,11 +27,11 @@ export const 유저태그 = Template.bind({});
   taggedUsers: [
     {
-      profileUrl: "?path=/story/components-image-userprofile--default",
+      userId: "0",
```
id는 숫자일텐대요~?

Ho2eny replied:
https://github.com/TheNewestWorld/groove-front/pull/60/commits/08c6271f0c875d7c16b3734b01a13b910b2310fd 반영했습니다~

Ho2eny committed:
Rename page CommunityComment to CommunityCommentList

Ho2eny committed:
feat: CommunityComment를 List와 Comment로 분리합니다

기존 CommunityComment Page 를 CommunityComment 컴포넌트와
CommunityCommentList 페이지로 분리합니다

Ho2eny committed:
userId의 타입을 변경합니다

Ho2eny committed:
더보기 이벤틀르 open과 close로 분리합니다

Ho2eny committed:
댓글 경과 시간을 받지 않고 생성 시간으로부터 계산하도록 수정
t

Ho2eny committed:
feat: 신고하기 기능 처리

Ho2eny commented:
@wlsdhr0831 진옥님 남겨주신 리뷰는 다 반영을 했습니다.
지금 다시 보기에는
1. `댓글 달기` 시에 UI 이벤트 -> 입력창에 `@ho2eny` 태그입력
2.  댓글이 없을 경우 빈 댓글 페이지 

이 2가지 요소들이 구현이 필요하다 생각되요.
현재 PR이 너무 커서 리뷰하기도 어렵다 보니 이 2가지는 커멘트 남겨두고 별도의 PR로 진행하려 합니다

Ho2eny committed:
TODO 커멘트

wlsdhr0831 commented on src/pages/CommunityCommentList/CommunityCommentList.view.tsx:
```diff
@@ -0,0 +1,100 @@
+import styles from "./CommunityCommentList.module.scss";
+
+import CommunityComment from "./components/CommunityComment";
+
+import CommunityCommentInput from "./components/CommunityCommentInput";
+
+import CommunityCommentItem, {
+  Props as CommunityCommentItemType,
+} from "./components/CommunityCommentItem";
+
+import Header from "../../components/Header";
+
+import { CloseIcon } from "../../assets/icon";
+import useOpen from "../../hooks/useOpen";
+import BottomSheetList from "../../components/BottomSheetList";
+import { useState } from "react";
```
사소한거지만! 임포트는 엔터 구분 없이 나열하고,
css관련 임포트 들이 하단에 배치되는 것이 좀 더 보편적인 것 같아요!

Ho2eny replied:
오 감사합니다! 그렇게 적용하겠습니다

Ho2eny replied:
1. React Component들 (CommunityCommentItem 등등)
2. React 관련 import (useState)
3. css 관련(style)

여기서도 혹시 순서가 있을까요..!?

wlsdhr0831 replied:
2,1,3 입니다!

wlsdhr0831 commented on src/pages/CommunityCommentList/CommunityCommentList.view.tsx:
```diff
@@ -0,0 +1,100 @@
+
+export interface Props {
+  comments: {
+    comment: {
+      commentId: number;
+      writer: string;
+      profileImage: string;
+      createdTime: Date;
+      comment: string;
+      taggedUsers?: { userId: number; nickname: string }[];
+    };
+    replies?: {
+      commentId: number;
+      writer: string;
+      profileImage: string;
+      createdTime: Date;
+      comment: string;
+      taggedUsers?: { userId: number; nickname: string }[];
+    }[];
```
여기서는 comment 타입과 replies 타입은 중복이 되는 것 같은데 type을 따로 정의해서 사용하시는건 어떠신가요?
그러면 따로 대조 없이 댓글과 리댓글은 똑같은 데이터 타입이구나 하고 바로 알 수 있을 것 같아요!

Ho2eny replied:
네 맞습니다, 이걸 정의하지를 못했네요..

wlsdhr0831 commented on src/pages/CommunityCommentList/CommunityCommentList.view.tsx:
```diff
@@ -0,0 +1,100 @@
+const CommunityCommentList = ({
+  comments = [],
+  onClickClose,
+  onClickCreateComment,
```
이거는 클릭이벤트라기보다 댓글 제출 이벤트 같은데
`onSubmitComment` 어떠신가요~?

wlsdhr0831 commented on src/pages/CommunityCommentList/CommunityCommentList.view.tsx:
```diff
@@ -0,0 +1,100 @@
+
+const CommunityCommentList = ({
+  comments = [],
+  onClickClose,
```
`onClose`라고 해도 좋을 것 같습니다!
닫기(close) 라는 행위에 무슨 액션을 취하는지 전달이 충분한것 같아요!
onClick~ 등의 타입을 쓰는 경우는 목적어 만을 가지고는 무엇을 해야하는지 전달이 모호할 때 쓰면 좋을 것 같습니다!
예를 들자면 이미지 클릭, 유저 아이디 클릭 등등..??

Ho2eny replied:
넵! 상세한 설명 감사합니다 👍

wlsdhr0831 commented on src/pages/CommunityCommentList/CommunityCommentList.view.tsx:
```diff
@@ -0,0 +1,100 @@
+  onClickReport,
+}: Props) => {
+  const { isOpen, onOpen, onClose } = useOpen();
+  const [reportId, setReportId] = useState<number>(-1);
```
왜 초기값을 -1로 하셨는지 물어봐도 될까요~?

Ho2eny replied:
초기 값은 ID에 나올 수 없는 값으로 설정하기 위해서 `-1`을 부여했는데요. 
문제가 없는 값으로 설정하는게 좋을까요??

wlsdhr0831 replied:
```
const NOT_SETTING = -1;
```
요렇게 그러면 상수하시는건 어떠세요~?

wlsdhr0831 commented on src/pages/CommunityCommentList/components/CommunityComment/CommunityComment.view.tsx:
```diff
@@ -0,0 +1,56 @@
+    <div>
+      <CommunityCommentItem {...comment} />
+
+      {replies &&
+        replies.length > 0 &&
```
이 두줄도 ? 사용으로 한줄에 가능하지 않나요~?

Ho2eny replied:
`?` 이 표현이 적용이 되지 않아서 그랬었습니다. 
`null`이 아닌 `undefined` 적용이 잘 안되서욥,,

wlsdhr0831 replied:
아하아하! 넵! 설명 감사합니당

wlsdhr0831 commented on src/pages/CommunityCommentList/components/CommunityCommentItem/CommunityCommentItem.view.tsx:
```diff
@@ -0,0 +1,97 @@
+  { time: "년", milliSeconds: 1000 * 60 * 60 * 24 * 365 },
+].reverse();
+
+const ConvertToElapsedTime = (createdTime: Date) => {
```
일반 함수는 소문자로 시작합니다~
대문자로 시작하는 함수는 JSX 컴포넌트를 의미함!

Ho2eny replied:
감사합니다!!

wlsdhr0831 commented on src/pages/CommunityCommentList/components/CommunityCommentItem/CommunityCommentItem.view.tsx:
```diff
@@ -0,0 +1,97 @@
+  );
+};
+
+const times: { time: string; milliSeconds: number }[] = [
+  { time: "분", milliSeconds: 1000 * 60 },
+  { time: "시간", milliSeconds: 1000 * 60 * 60 },
+  { time: "일", milliSeconds: 1000 * 60 * 60 * 24 },
+  { time: "개월", milliSeconds: 1000 * 60 * 60 * 24 * 30 },
+  { time: "년", milliSeconds: 1000 * 60 * 60 * 24 * 365 },
+].reverse();
+
+const ConvertToElapsedTime = (createdTime: Date) => {
+  const now = new Date().getTime();
+  const diff = now - createdTime.getTime();
+
+  for (const value of times) {
+    const elapsedTime = Math.floor(diff / value.milliSeconds);
+    if (elapsedTime > 0) {
+      return `${elapsedTime}${value.time}전`;
+    }
+  }
+
+  return "방금 전";
+};
```
외부 파일의 함수로 빼도 좋을 것 같습니다~
해당 폴더의 helpers.ts 라던가..??

Ho2eny replied:
`src`의 `helper` 쪽에 추가해 놓겠습니다

wlsdhr0831 replied:
앗 아니요 해당 컴포넌트에서만 사용하는거라면 해당 페이지 디렉토리 하단에 있으면 됩니다~!

wlsdhr0831 commented on src/pages/CommunityCommentList/components/CommunityComment/CommunityComment.view.tsx:
```diff
@@ -0,0 +1,56 @@
+          className={styles.readMore}
+          isOpen={readMore.isOpen}
+          replyCount={replies!.length}
+          onClickOpen={() => {
+            setReadMore({
+              isOpen: true,
+              limit: readMore.limit,
+            });
+          }}
+          onClickClose={() => {
+            setReadMore({
+              isOpen: false,
+              limit: readMore.limit,
```
이거 limit 값은 변동이 없는데 작성이 되어있는건가요~?
그냥 useOpen hook 써서 true, false 만 컨트롤을 하면 되는게 아닌지요..??

Ho2eny replied:
네, 이 부분을 기획분들께 질문을 드릴까 고민하다가 잠깐 넘겼었는데, 생각을 못했었네요. 
그냥 `limit` 없이 다 보여주는게 좋을까요?

wlsdhr0831 replied:
limit 이 필요하다 해도 해당 컴포넌트의 props 값으로 주면 되지 않을까요~?
어차피 폇다 안폇다 이거고, 버튼을 누를때마다 10, 20, 30 이렇게 늘어나는게 아니니까~?

Ho2eny committed:
리뷰를 반영합니다

- import 순서 변경
- 공통 타입 정의
- rename 이벤트

wlsdhr0831 reviewed (approved):
👍 👍
